src/components/sequencing/Sequences.svelte

```diff
@@ -0,0 +1,172 @@
+<svelte:options immutable={true} />
+
+<script lang="ts">
+  import { goto } from '$app/navigation';
+  import { base } from '$app/paths';
+  import { userSequences, userSequencesColumns } from '../../stores/sequencing';
+  import effects from '../../utilities/effects';
+  import { compare } from '../../utilities/generic';
+  import Input from '../form/Input.svelte';
+  import Chip from '../ui/Chip.svelte';
+  import CssGrid from '../ui/CssGrid.svelte';
+  import CssGridGutter from '../ui/CssGridGutter.svelte';
+  import DataGrid from '../ui/DataGrid/DataGrid.svelte';
+  import DataGridActions from '../ui/DataGrid/DataGridActions.svelte';
+  import Panel from '../ui/Panel.svelte';
+  import SequenceEditor from './SequenceEditor.svelte';
+
+  type CellRendererParams = {
+    deleteSequence: (sequence: UserSequence) => void;
+    editSequence: (sequence: UserSequence) => void;
+  };
+  type SequencesCellRendererParams = ICellRendererParams<UserSequence> & CellRendererParams;
+
+  const columnDefs: DataGridColumnDef[] = [
+    {
+      field: 'id',
+      headerName: 'ID',
+      resizable: true,
+      sortable: true,
+      suppressAutoSize: true,
+      suppressSizeToFit: true,
+      width: 100,
+    },
+    { field: 'name', headerName: 'Name', resizable: true, sortable: true },
+    { field: 'authoring_command_dict_id', headerName: 'Command Dictionary ID', resizable: true, sortable: true },
+    { field: 'created_at', headerName: 'Created At', resizable: true, sortable: true },
+    { field: 'updated_at', headerName: 'Created At', resizable: true, sortable: true },
+    {
+      cellClass: 'action-cell-container',
+      cellRenderer: (params: SequencesCellRendererParams) => {
+        const actionsDiv = document.createElement('div');
+        actionsDiv.className = 'actions-cell';
+        new DataGridActions({
+          props: {
+            deleteCallback: params.deleteSequence,
+            deleteTooltip: {
+              content: 'Delete Sequence',
+              placement: 'bottom',
+            },
+            editCallback: params.editSequence,
+            editTooltip: {
+              content: 'Edit Sequence',
```

Comment:
Not sure if this is expected / normal usage but was seeing a lot of slowness when trying to edit a sequence after queuing up lots of sequence previews

https://user-images.githubusercontent.com/4419607/187779096-290bdbc6-9a1b-4271-bc2e-630980ff274e.mov

Reply:
Good catch. The `fetch` action for Seq JSON is slightly resource heavy. The backend takes on average 5s to process a Clipper-sized dictionary. Will add an `AbortController` to the request so we can stop the fetch when changing your selection.

Reply:
Fixed